crates/agglayer-interop-types/src/aggchain_proof/mod.rs

```diff
@@ -26,7 +26,9 @@ pub enum AggchainData {
         /// Optional aggchain proof public values.
         public_values: Option<Box<AggchainProofPublicValues>>,
     },
-    MultisigOnly(MultisigPayload),
+    MultisigOnly {
+        multisig: MultisigPayload,
+    },
```

Comment:
Note, since it might not be obvious: this is the actual fix. The `#[serde(flatten)]` attribute, used in `Certificate`, does not play nice with tuple variants. It requires the contents to be a struct or a struct-variant.